pkg/cli/cmd/radinit/recipe.go

```diff
@@ -52,148 +127,57 @@ func NewDevRecipeClient() DevRecipeClient {
 func (drc *devRecipeClient) GetDevRecipes(ctx context.Context) (map[string]map[string]corerp.RecipePropertiesClassification, error) {
-	reg, err := remote.NewRegistry(DevRecipesRegistry)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create client to registry %s -  %s", DevRecipesRegistry, err.Error())
-	}
+	logger := ucplog.FromContextOrDiscard(ctx)
 
 	// The tag will be the major.minor version of the release.
 	tag := version.Channel()
 	if version.IsEdgeChannel() {
 		tag = "latest"
 	}
 
-	// Temporary solution to get all repositories.
-	// The issue is that if RepositoryListPageSize is not specified the default is 100.
-	// We have 104 repositories in the registry as of 12 Oct 2023. That is why processRepositories
-	// function was being called twice and the second call was overwriting all the recipes.
-	// TODO: Remove this once we have a better solution.
-	reg.RepositoryListPageSize = 1000
-
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// if repository has the correct path it should look like: <registryPath>/recipes/<category>/<type>:<tag>
-	// Ex: ghcr.io/radius-project/recipes/local-dev/rediscaches:0.20
-	// The start parameter is set to "radius-rp" because our recipes are after that repository.
-	err = reg.Repositories(ctx, "", func(repos []string) error {
-		// validRepos will contain the repositories that have the requested tag.
-		validRepos := []string{}
-		for _, repo := range repos {
-			r, err := reg.Repository(ctx, repo)
-			if err != nil {
-				continue
-			}
-
-			tagExists := false
-			err = r.Tags(ctx, "", func(tags []string) error {
-				for _, t := range tags {
-					if t == tag {
-						tagExists = true
-						break
-					}
-				}
-				return nil
-			})
-			if err != nil {
-				continue
-			}
-
-			if tagExists {
-				validRepos = append(validRepos, repo)
-			}
-		}
-
-		recipes = processRepositories(validRepos, tag)
-		return nil
-	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to list recipes available in registry at path  %s -  %s", DevRecipesRegistry, err.Error())
-	}
-
-	return recipes, nil
-}
-
-// processRepositories processes the repositories and returns the recipes.
-func processRepositories(repos []string, tag string) map[string]map[string]corerp.RecipePropertiesClassification {
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// We are using the default recipe.
-	name := "default"
-
-	for _, repo := range repos {
-		// Skip dev environment recipes.
-		// dev repositories is in the form of ghcr.io/radius-project/dev/recipes/local-dev/secretstores:latest
-		// We should skip the dev repositories.
-		if isDevRepository(repo) {
+	validDevRecipes := map[string]map[string]corerp.RecipePropertiesClassification{}
+	for _, devRecipe := range availableDevRecipes() {
+		repo, err := remote.NewRepository(devRecipe.RepoPath)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("failed to create client to repository %s", devRecipe.RepoPath))
```

Comment:
Can you explain why we should do this instead of returning an error? Continuing in the face of an error should not be a common practice.

Reply:
The reason we ignore an error here is that this is a part of the `rad init` flow and I believe we shouldn't block the `rad init` flow because of an error in one of the dev recipes. I am not sure but I think I remember coming to this decision with the team in the very first iteration of the dev recipes implementation. Please let me know if this should terminate the `rad init` flow and I can update the PR accordingly. This is the same as the current behavior on main: https://github.com/radius-project/radius/blob/main/pkg/cli/cmd/radinit/recipe.go#L83. cc/ @AaronCrawfis @rynowak